flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -210,6 +207,7 @@ public void testSetupApplicationMasterContainer() {
 				.getCommands().get(0));
 
 		// logback + log4j, with/out krb5, different JVM opts
+		// IMPORTANT: Beaware that we are using side effects here to modify the created YarnClusterDescriptor
```

Comment:
We keep a reference on the `Configuration` object being held by the `ClusterDescriptor`. Thus, the different settings for the different assert statement end up in the same configuration instance. I'll add this to the comment.